### PR TITLE
Withdraw Assets Before Checking Out in `OnReapIdentity` impl

### PR DESCRIPTION
**Pull request by joepetrowski**
Follow up to fix a bug from #1814 discovered in XCM emulator testing.

I mistakenly thought that checking out an asset would withdraw it from the sender. This actually withdraws the asset before checking out.

### CODE REVIEW

**Comment by 0xmovses** [comment]
Could you change the base branch to `joe-people-chain`? So that I can merge this in to my PR.

File: polkadot/runtime/westend/src/impls.rs [hunk 1]
```diff
@@ -100,6 +100,12 @@ where
 		// Do `check_out` accounting since the XCM Executor's `InitiateTeleport` doesn't support
 		// unpaid teleports.
 
+		// withdraw the asset from `who`
+		let who_origin =
+			Junction::AccountId32 { network: None, id: who.clone().into() }.into_location();
+		let _withdrawn = xcm_config::LocalAssetTransactor::withdraw_asset(&wnd, &who_origin, None)
+			.map_err(|_| pallet_xcm::Error::<Runtime>::CannotWithdraw)?;
```
Comment (0xmovses) [inline polkadot/runtime/westend/src/impls.rs:107]: The short hands `wnd` and `roc` always require me to do a double take when reading the code. If this isn't a house style I'd prefer to have the full names `westend` and `rococo`, but up to you.

**Review by 0xmovses — APPROVED** [review]
Just one nit.

File: polkadot/runtime/westend/src/impls.rs (hunk 1 above)
Comment (joepetrowski) [reply polkadot/runtime/westend/src/impls.rs:107]: "Westend" and "Rococo" are the names of the networks, "WND" and "ROC" are the names of their native asset, respectively.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4548403

File: polkadot/runtime/rococo/src/impls.rs [hunk 2]
```diff
@@ -100,6 +100,12 @@ where
 		// Do `check_out` accounting since the XCM Executor's `InitiateTeleport` doesn't support
 		// unpaid teleports.
 
+		// withdraw the asset from `who`
+		let who_origin =
+			Junction::AccountId32 { network: None, id: who.clone().into() }.into_location();
+		let _withdrawn = xcm_config::LocalAssetTransactor::withdraw_asset(&roc, &who_origin, None)
```
Comment (franciscoaguirre) [inline polkadot/runtime/rococo/src/impls.rs:106]: Can we also `log::error!` the actual error before raising the new error? For better logging and debugging

File: polkadot/runtime/westend/src/impls.rs (hunk 1 above)
Comment (acatangiu) [inline polkadot/runtime/westend/src/impls.rs:107]:
```suggestion
			.map_err(|err| {
				log::error!(
					target: "runtime::on_reap_identity",
					"withdraw_asset(what: {:?}, who_origin: {:?}) error: {:?}",
					wnd, who_origin, err
				);
				pallet_xcm::Error::<Runtime>::CannotWithdraw
			})?;

**Review by acatangiu — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

File: polkadot/runtime/rococo/src/impls.rs [hunk 3]
```diff
@@ -100,14 +100,34 @@ where
...
```
[the remark this answers is not in the document]
Comment (joepetrowski) [reply polkadot/runtime/rococo/src/impls.rs:106]: You mean burned when it exits scope? But the `check_out` should mint this asset into the `CheckingAccount`, so the burn should be offset by that.

**Pull request merged**